apps/shell/utils/helpers.js

```diff
@@ -1,10 +1,67 @@
-const minimatch = require('minimatch')
+const minimatch = require('minimatch');
+const glob      = require('glob');
+const path      = require('path');
+const yaml      = require('js-yaml');
+const fs        = require('fs');
 
 function hostInAllowList(allowlist, host) {
   allowlist = Array.from(allowlist);
   return allowlist.some((value) => minimatch(host, value))
 }
 
+/**
+ * If the url has a match with host_path_rx, the respective capture groups are returned
+ * 
+ * @param {string} url request url
+ * @param {string} default_sshhost returned hostname if default
+ * 
+ * @return {Array}
+ */
+function hostAndDirFromURL(url, default_sshhost){
+  const host_path_rx = '/ssh/([^\\/\\?]+)([^\\?]+)?(\\?.*)?$';
+  let match = url.match(host_path_rx), 
+  hostname = null, 
+  directory = null;
+
+  if (match) {
+    hostname = match[1] === "default" ? default_sshhost : match[1];
+    directory = match[2] ? decodeURIComponent(match[2]) : null;
+  }
+  return [hostname, directory];
+}
+
+class HostAllowlist{
```

Comment:
I think using a class here is okay but it needs to be in its own file and imported as a module around the application.

Reply:
If `HostAllowlist` is going to be a class, then there should be methods to check if a host is in the allowlist, etc. You shouldn't have to import functionality from the helpers file to make use of the class. It should be able to be used independently.

Reply:
`this.allowlist` is a Set and we can modify it directly but its better to do it through a function in the class like this:

```js
function addToAllowlist(host) {
  this.allowlist.add(host)
}
```